kaitai/kaitaistream.h

```diff
@@ -281,6 +282,13 @@ inline auto to_signed(From from) {
     throw std::runtime_error("toSigned: Invalid casting");
 }
 
+template<class Output, class Read>
+inline void type_pun(Output& output, Read& read)
+{
+    static_assert(sizeof(output) == sizeof(output), "Type sizes don't match");
```

Comment:
static_assert(sizeof(read) == sizeof(output), "Type sizes don't match");

Reply:
Telltale sign that I'm too tired, thanks for spotting it.

Reply:
haha don't worry!